snapshot/snapshot.go

```diff
@@ -435,6 +435,7 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 
 	// Local snapshot setup failed. Generally means something critical has gone wrong.
 	log.G(lCtx).WithError(err).Warnf("failed to prepare local snapshot; %v", ErrDeferToContainerRuntime)
+	commonmetrics.IncOperationCount(commonmetrics.LocalMountFailureCount, digest.Digest(""))
 	return mounts, nil
```

Comment:
Do you know why when a critical error occurs we return nil error in this call path?

Reply:
This is a thing most remote snapshotters do. Whenever we successfully do a prepare call you can see we return ErrAlreadyExists, so that containerd doesn't try to pull it on its own. It will only try to pull an image if there is no error preparing the snapshot, so by returning nil we are signaling to containerd that it needs to pull this layer.

Reply:
Gotcha thanks for clarifying.